avocado/utils/linux_modules.py

```diff
@@ -119,7 +119,12 @@ def loaded_module_info(module_name):
                 elif key == 'depends':
                     value = items[1].split(',')
                 elif key == 'parm':
-                    param_list.append(items[1].split(':')[0])
+                    param_dic = {'type': None}
+                    param_dic['name'] = items[1].split(':')[0]
+                    param_type = re.search(r"\((\w+)\)", items[-1])
+                    if param_type is not None:
+                        param_dic['type'] = param_type.group(1)
+                    param_list.append(param_dic)
```

Comment:
Well the only problem I have here is that it again changes the format of the returned info:

Old:
```
{'depends': ['drm_kms_helper', 'drm', 'video', 'i2c-algo-bit'],
 'filename': '/lib/modules/4.9.13-201.fc25.x86_64/kernel/drivers/gpu/drm/i915/i915.ko.xz',
 'name': 'i915',
 'params': ['modeset',
            'panel_ignore_lid',
            'semaphores',
            'enable_rc6',
            'enable_dc',
            'enable_fbc',
            'lvds_channel_mode',
...
```

New
```
{'depends': ['drm_kms_helper', 'drm', 'video', 'i2c-algo-bit'],
 'filename': '/lib/modules/4.9.13-201.fc25.x86_64/kernel/drivers/gpu/drm/i915/i915.ko.xz',
 'name': 'i915',
 'params': [{'name': 'modeset', 'type': 'int'},
            {'name': 'panel_ignore_lid', 'type': 'int'},
            {'name': 'semaphores', 'type': 'int'},
            {'name': 'enable_rc6', 'type': 'int'},
            {'name': 'enable_dc', 'type': 'int'},
            {'name': 'enable_fbc', 'type': 'int'},
            {'name': 'lvds_channel_mode', 'type': 'int'},
            {'name': 'lvds_use_ssc', 'type': 'int'},
...
```
And this change is bigger then the previous one. Anyway I hope you are the only user of this library and you do try to stabilize the results soon as when we have people depending on those we'd have to start dealing with the backward compatibility issues. Please do consider that when designing future utils...